crates/wasm-sdk/src/errors.rs

```diff
@@ -15,47 +11,98 @@ pub enum ParseCounterError {
     Utf8(#[from] std::str::Utf8Error),
 }
 
-/// Returned when there was an issue getting or setting a parameter.
+/// Returned when there is an issue with the environment variable requested by the plugin.
 #[derive(thiserror::Error, Debug)]
-pub enum ParamError {
+pub enum EnvVarError {
+    #[error("access to environment variable '{var}' denied")]
+    Permission { var: String },
+    #[error("environment variable '{var}' missing")]
+    Missing { var: String },
     #[error("{message}")]
-    Json { message: String },
+    InvalidUnicode { message: String },
+    #[error("an unexpected error occurred accessing environment variable '{var}'")]
+    Other { var: String },
 }
 
-impl From<crate::bulwark_host::ParamError> for ParamError {
-    fn from(error: crate::bulwark_host::ParamError) -> Self {
+impl From<crate::wit::bulwark::plugin::config::Error> for EnvVarError {
+    fn from(error: crate::wit::bulwark::plugin::config::Error) -> Self {
         match error {
-            crate::bulwark_host::ParamError::Json(message) => ParamError::Json { message },
+            crate::wit::bulwark::plugin::config::Error::Permission(var) => {
+                EnvVarError::Permission { var }
+            }
+            crate::wit::bulwark::plugin::config::Error::Missing(var) => {
+                EnvVarError::Missing { var }
+            }
+            crate::wit::bulwark::plugin::config::Error::InvalidUnicode(var) => {
+                EnvVarError::InvalidUnicode {
+                    message: format!("environment variable '{var}' contained invalid unicode"),
+                }
+            }
+            crate::wit::bulwark::plugin::config::Error::InvalidSerialization(_) => {
+                // This shouldn't happen because environment variables are always returned as bytes or strings
+                unreachable!()
+            }
+            crate::wit::bulwark::plugin::config::Error::InvalidConversion(_) => {
+                // This shouldn't happen because environment variables are always returned as bytes or strings
+                unreachable!()
+            }
+        }
+    }
```

Comment:
Two options:
1. Avoid panics and hopefully recover gracefully but then likely fail to discover that an assumption was incorrect because the error message got buried in logs.
2. Hopefully never panic, but then if the assumption does prove incorrect, you're a lot more likely to discover it because a panic is much noisier.

I prefer the latter, particularly once the test suite grows a bit more robust and can adequately cover enough of the code paths to lend confidence that assumptions are correct.